pkg/datarepair/checker/checker.go

```diff
@@ -120,12 +122,39 @@ func lookupResponsesToNodes(responses *pb.LookupResponses) []*pb.Node {
 	return nodes
 }
 
-// Run
-func (c *checker) Run() error {
-	return nil
+// Run the checker loop
+func (c *checker) Run(ctx context.Context, interval time.Duration) error {
+	zap.S().Info("Checker is starting up")
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	var errs []error
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				zap.S().Info("Starting segment checker service")
+				go func() {
+					err := c.IdentifyInjuredSegments(ctx)
+					if err != nil {
+						errs = append(errs, err)
+					}
+				}()
+				return
```

Comment:
this will stop the loop after the first tick, is that what you want?

Reply:
nope I don't want that. rewriting.